azure-kusto-data/azure/kusto/data/_models.py

```diff
@@ -48,31 +50,29 @@ def __init__(self, columns, row):
                     self._hidden_values.append(value)
                 continue
 
-            if column_type in ["datetime", "timespan", "decimal"]:
-                if value is None:
-                    typed_value = None
-                    if HAS_PANDAS:
-                        self._hidden_values.append(None)
-                else:
-                    # If you are here to read this, you probably hit some datetime/timedelta inconsistencies.
-                    # Azure-Data-Explorer(Kusto) supports 7 decimal digits, while the corresponding python types supports only 6.
-                    # One example why one might want this precision, is when working with pandas.
-                    # In that case, use azure.kusto.data.helpers.dataframe_from_result_table which takes into account the original value.
-                    typed_value = KustoResultRow.convertion_funcs[column_type](value)
-
-                    # this is a special case where plain python will lose precision, so we keep the precise value hidden
-                    # when transforming to pandas, we can use the hidden value to convert to precise pandas/numpy types
-                    if HAS_PANDAS:
-                        if column_type == "datetime":
-                            self._hidden_values.append(to_pandas_datetime(value))
-                        if column_type == "timespan":
-                            self._hidden_values.append(to_pandas_timedelta(value, typed_value))
-                        if column_type == "decimal":
-                            self._hidden_values.append(value)
+            if value is None:
```

Comment:
how about you get rid of the if else...

```python
# If you are here to read this, you probably hit some datetime/timedelta inconsistencies.
# Azure-Data-Explorer(Kusto) supports 7 decimal digits, while the corresponding python types supports only 6.
# One example why one might want this precision, is when working with pandas.
# In that case, use azure.kusto.data.helpers.dataframe_from_result_table which takes into account the original value.
typed_value = (
    KustoResultRow.convertion_funcs[column_type](value)
    if column_type in KustoResultRow.convertion_funcs and value is not None
    else value
)

    # This is a special case where plain python will lose precision, so we keep the precise value hidden.
# When transforming to pandas, we can use the hidden value to convert to precise pandas/numpy types
if HAS_PANDAS:
    self._hidden_values.append(
        KustoResultRow.pandas_funcs[column_type](value, typed_value)
        if column_type in KustoResultRow.pandas_funcs and value is not None
        else typed_value 
    )
```

Reply:
This was my initial implementation (see [commit](https://github.com/Azure/azure-kusto-python/blob/564b972bd36e453fd87c43092f1263b0d7af28f9/azure-kusto-data/azure/kusto/data/_models.py#L61)), but I think this way is cleaner (and more efficient with regards to function calls). Let me know what you think.

Reply:
How is it more efficient? you can always reverse 
`if value is not None and column_type in KustoResultRow.pandas_funcs`
which will do exactly the same. 

My problem with the `if value is Non` is that it doesn't convey the intention of the code as well as the code without it..

But, it's not that big of a deal to me.. if you like it the way it was, ok.